builtin/providers/aws/resource_aws_dynamodb_table.go

```diff
@@ -336,6 +366,32 @@ func resourceAwsDynamoDbTableUpdate(d *schema.ResourceData, meta interface{}) er
 		waitForTableToBeActive(d.Id(), meta)
 	}
 
+	if d.HasChange("stream_specification") {
+		log.Printf("[DEBUG] Changed Stream Specification")
+		req := &dynamodb.UpdateTableInput{
+			TableName: aws.String(d.Id()),
+		}
+
+		vs := d.Get("stream_specification").(*schema.Set).List()
+		if len(vs) > 0 {
+			spec := vs[0].(map[string]interface{})
+			streamSpecification := &dynamodb.StreamSpecification{
+				StreamEnabled:   aws.Bool(bool(spec["enabled"].(bool))),
+				StreamViewType: aws.String(string(spec["view_type"].(string))),
+			}
+			req.StreamSpecification = streamSpecification
+			d.SetPartial("stream_specification")
+		}
+
+		_, err := dynamodbconn.UpdateTable(req)
```

Comment:
It seems like there are several distinct calls to the same ``UpdateTable`` function in here. This has been going on long before your change of course, but I wonder why it's written this way... the other resources I've seen in Terraform tend to gather all of the updates into a single request. Is there something weird about the DynamoDB API that makes this approach desirable vs. just building a single ``UpdateTableInput`` struct across this whole function and submitting it in a single call?